### ISSUE TITLE
How about the sliding DFT?

### ISSUE THREAD

**Issue by jurihock** [jurihock/stftPitchShift#5]
Evaluate following work:

* [Sliding is smoother than jumping](https://quod.lib.umich.edu/cgi/p/pod/dod-idx/sliding-is-smoother-than-jumping.pdf?c=icmc;idno=bbp2372.2005.086;format=pdf)
* [The Sliding Phase Vocoder](https://quod.lib.umich.edu/cgi/p/pod/dod-idx/sliding-phase-vocoder.pdf?c=icmc;idno=bbp2372.2007.198;format=pdf)
* [Understanding and Implementing the Sliding DFT](https://www.dsprelated.com/showarticle/776.php)
* [bronsonp/SlidingDFT](https://github.com/bronsonp/SlidingDFT)

**Comment by jurihock** [comment]
The sliding DFT will be first implemented and tested out in [voyx](https://github.com/jurihock/voyx).

**Comment by jurihock** [comment]
As mentioned in [3] the classic SDFT implementation suffers from accumulated frequency bin errors. These frequency bin errors may become audible after about 10 minutes on single precision, depending on the source signal e.g. cyclic chirp.

The mSDFT implementation proposed in [6] offers reduced error level even on single precision, but at "slightly" higher computational cost (something between 1.5 and 2).

**Comment by jurihock** [comment]
The author of [11] reports a lower numerical error level of *gSDFT* compared to *mSDFT* from [6], but also a higher workload on larger windows. The finally proposed *oSDFT* should have exactly the same numerical error level as *gSDFT* and the lowest workload "among the existing stable" SDFT algorithms. We will see...

**Comment by jurihock** [comment]
TODO: The SDFT code is consolidated in https://github.com/jurihock/sdft, so use it in the future...

**Comment by NathanRoyer** [comment]
Hi, first of all thank you for publishing this project, the python implementation is very clear and I was able to understand your algorithm pretty well.

I see that you have "removed the SDFT prototype" just above.
Have you given up on using the SDFT in this project? If yes, could you give the reason?

Thanks

**Comment by jurihock** [comment]
Hi Nathan,

the main reason for investigating the SDFT was the ability to reduce the algorithmic latency of the "classic" STFT procedure. Meanwhile I've discovered the properties of the asymmetric windows #38, which allows to lower the algorithmic latency of the "classic" STFT as well. Even if this can lead to artefacts under certain circumstances #44.

In terms of "quality", I believe that the QDFT would be a more suitable tool than the SDFT (and also STFT), due to its logarithmic resolution. It's likely that both QDFT and SDFT approaches would require some major algorithmic redesign. Especially for formant preservation. Real-time processing is another key consideration. Both QDFT and SDFT are definitely more computationally intensive than STFT, due to the sample-by-sample data processing.

In short, the SDFT is not currently in focus. I think QDFT or maybe TFR #43 would make more sense for this kind of application.
